mlog/testing.go

```diff
@@ -23,9 +25,10 @@ func (tw *testingWriter) Write(b []byte) (int, error) {
 
 // NewTestingLogger creates a Logger that proxies logs through a testing interface.
 // This allows tests that spin up App instances to avoid spewing logs unless the test fails or -verbose is specified.
-func NewTestingLogger(tb testing.TB) *Logger {
+func NewTestingLogger(tb testing.TB, buffer *bytes.Buffer) *Logger {
 	logWriter := &testingWriter{tb}
-	logWriterSync := zapcore.AddSync(logWriter)
+	multiWriter := io.MultiWriter(logWriter, buffer)
```

Comment:
Would it make sense to leave the `io.MultiWriter` in `setupTestHelper`? It seems to clutter the definition of the testing logger when it's the caller who owns the buffer anyway.

Reply:
You have to have the `MultiWriter` to create a `Logger`.
To create `MultiWriter` you have to have `testingWriter`.
We can create `MultiWriter` in `setupTestHelper` and pass it to the `NewTestingLogger`, but we would need `testingWriter` making public and it does not make sense to do that.

Reply:
Ah, good point -- I hadn't fully thought through the consequences. In that case, can we make this method accept an `io.Writer` instead of a `*bytes.Buffer` to more clearly emphasize the ownership?